types/GasPriceOracle.md

```diff
@@ -0,0 +1,128 @@
+To generate a query ID, get the `bytes32` value of the `keccak` hash of the query data (defined above). For example, in Solidity:
+```s
+keccak256(queryData)
+```
+
+You can use [this tool](https://queryidbuilder.herokuapp.com/custom) to generate query IDs.
+
+
+## JSON Representation
+The JSON representation of your new query type is needed to construct query objects in a variety of languages. It contains the essential components of your query: type name, parameters in an ordered list and their corresponding value types, as well as the expected response type for the query.
+
+For example, the JSON representation of a `SpotPrice` query:
+```json
+{
+    "type": "GasPriceOracle",
+    "abi": [
+        {
+            "type": "uint256",
+            "name": "chainId",
+        },
+        {
+            "type": "uint256",
+            "name": "timestamp",
+        },
+    ],
+    "response": {
+        "type": "ufixed256x18",
+        "packed": false,
+    }
+}
+```
+
+
+## Example
+A working example mapping of all the various inputs and parameters to a valid queryID. 
+
+The queryData for mainnet ethereum at unix timestamp `1650465649`:
+
+```s
+queryData = abi.encode("GasPriceOracle", abi.encode("1", "1650465649"))
+queryId = keccack256(queryData)
+```
+
+queryData: `0x00000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000e47617350726963654f7261636c65000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000062601b71`
+
+queryId:
+`0x7dd07b75e8c096ea1926e9a2cc8f749b04581d0ae88a892989f05790384344f6`
+
+You can use [this tool](https://queryidbuilder.herokuapp.com/custom) to generate query IDs.
+
+
+## Dispute Considerations
+
+Note that following this guide does not prevent you from being disputed or guarantee reporters will properly put a value on-chain. Tellor is decentralized.  This repo is a start to the education necessary for a fully decentralized oracle, but please focus on communication and working with reporters to prevent unneccesary disputes and at the same time encourage monitoring and punishment of bad data. 
+
+- retreiving a gas price from one historical transaction is not recommended. gas price is very volatile
+- retreiving the current gas price from metamask is not recommended. gas price is very volatile
+- care should be used when reporting from black box aggregators
+- use EIP-1559 gas strategy
+
+
+
+## Suggested Data Sources
+
+Where can reporters retrieve the query's response? Are there APIs available so reporters can programmatically retrieve the data for your query's expected response? Are there multiple possible sources for the expected response?
+
+Owlracle privode's a "gas price history" API. It supports the following networks:
```

Comment:
Typos. *The GasPriceOracle query type provides

Reply:
fixed the typo